coderd/database/dbauthz/dbauthz_test.go

```diff
@@ -3698,835 +3698,716 @@ func (s *MethodTestSuite) TestCryptoKeys() {
+		at := dbtime.Now()
+		dbm.EXPECT().GetWorkspaceBuildStatsByTemplates(gomock.Any(), at).Return([]database.GetWorkspaceBuildStatsByTemplatesRow{}, nil).AnyTimes()
+		check.Args(at).Asserts(rbac.ResourceSystem, policy.ActionRead)
 	}))
-	s.Run("GetWorkspaceAgentScriptTimingsByBuildID", s.Subtest(func(db database.Store, check *expects) {
-		dbtestutil.DisableForeignKeysAndTriggers(s.T(), db)
-		workspace := dbgen.Workspace(s.T(), db, database.WorkspaceTable{})
-		job := dbgen.ProvisionerJob(s.T(), db, nil, database.ProvisionerJob{
-			Type: database.ProvisionerJobTypeWorkspaceBuild,
-		})
-		build := dbgen.WorkspaceBuild(s.T(), db, database.WorkspaceBuild{JobID: job.ID, WorkspaceID: workspace.ID})
-		resource := dbgen.WorkspaceResource(s.T(), db, database.WorkspaceResource{
-			JobID: build.JobID,
-		})
-		agent := dbgen.WorkspaceAgent(s.T(), db, database.WorkspaceAgent{
-			ResourceID: resource.ID,
-		})
-		script := dbgen.WorkspaceAgentScript(s.T(), db, database.WorkspaceAgentScript{
-			WorkspaceAgentID: agent.ID,
-		})
-		timing := dbgen.WorkspaceAgentScriptTiming(s.T(), db, database.WorkspaceAgentScriptTiming{
-			ScriptID: script.ID,
-		})
-		rows := []database.GetWorkspaceAgentScriptTimingsByBuildIDRow{
-			{
-				StartedAt:          timing.StartedAt,
-				EndedAt:            timing.EndedAt,
-				Stage:              timing.Stage,
-				ScriptID:           timing.ScriptID,
-				ExitCode:           timing.ExitCode,
-				Status:             timing.Status,
-				DisplayName:        script.DisplayName,
-				WorkspaceAgentID:   agent.ID,
-				WorkspaceAgentName: agent.Name,
-			},
-		}
-		check.Args(build.ID).Asserts(rbac.ResourceSystem, policy.ActionRead).Returns(rows)
+	s.Run("UpsertNotificationReportGeneratorLog", s.Mocked(func(dbm *dbmock.MockStore, _ *gofakeit.Faker, check *expects) {
+		arg := database.UpsertNotificationReportGeneratorLogParams{NotificationTemplateID: uuid.New(), LastGeneratedAt: dbtime.Now()}
+		dbm.EXPECT().UpsertNotificationReportGeneratorLog(gomock.Any(), arg).Return(nil).AnyTimes()
+		check.Args(arg).Asserts(rbac.ResourceSystem, policy.ActionCreate)
 	}))
-	s.Run("DisableForeignKeysAndTriggers", s.Subtest(func(db database.Store, check *expects) {
+	s.Run("GetProvisionerJobTimingsByJobID", s.Mocked(func(dbm *dbmock.MockStore, faker *gofakeit.Faker, check *expects) {
+		j := testutil.Fake(s.T(), faker, database.ProvisionerJob{Type: database.ProvisionerJobTypeWorkspaceBuild})
+		b := testutil.Fake(s.T(), faker, database.WorkspaceBuild{JobID: j.ID})
+		ws := testutil.Fake(s.T(), faker, database.Workspace{ID: b.WorkspaceID})
+		dbm.EXPECT().GetProvisionerJobByID(gomock.Any(), j.ID).Return(j, nil).AnyTimes()
+		dbm.EXPECT().GetWorkspaceBuildByJobID(gomock.Any(), j.ID).Return(b, nil).AnyTimes()
+		dbm.EXPECT().GetWorkspaceByID(gomock.Any(), b.WorkspaceID).Return(ws, nil).AnyTimes()
+		dbm.EXPECT().GetProvisionerJobTimingsByJobID(gomock.Any(), j.ID).Return([]database.ProvisionerJobTiming{}, nil).AnyTimes()
```

Comment:
This doesn't return any of the objects which the test is purporting to validate.

Reply:
Seems like it's not needed since we use the job to authz the call 👍